compiler/plugins/input/Torch/torch-iree/PluginRegistration.cpp

```diff
@@ -93,16 +93,35 @@ struct TorchSession
     const Dialect *torchConversionDialect = ctx->getLoadedDialect("torch_c");
     const Dialect *tmTensorDialect = ctx->getLoadedDialect("tm_tensor");
 
+    bool hasTorch = false;
+    bool hasOnnx = false;
+
     module.walk([&](Operation *op) {
       Dialect *d = op->getDialect();
       if (d == torchDialect || d == torchConversionDialect) {
-        typeMnemonics.insert("torch");
+        hasTorch = true;
       } else if (d == tmTensorDialect) {
         // TODO: Retire the tm_tensor input pipeline
         typeMnemonics.insert("tm_tensor");
       }
       return WalkResult::advance();
     });
+
+    for (auto funcOp : module.getOps<func::FuncOp>()) {
+      if (funcOp->getAttrOfType<mlir::IntegerAttr>(
+              "torch.onnx_meta.opset_version")) {
+        hasOnnx = true;
+        break;
+      }
+    }
+
+    // ONNX is considered a superset of Torch. It runs all of the Torch
+    // pipelines with an extra ONNX-specific preprocessing step.
+    if (hasOnnx) {
+      typeMnemonics.insert("onnx");
+    } else if (hasTorch) {
```

Comment:
This should not be `else if` correct? It could contain both `torch` and `onnx` ops (and the `onnx` lowerings require the torch support as well).

Reply:
See the code above and the comment here. The "onnx" type is a superset of the "torch" type. It runs onnx passes _and_ torch passes. We don't currently have a way to run multiple input type conversions, so this ensures that onnx (torch + onnx) is run and _not_ the base (torch).

Reply:
This is also how the "stablehlo" and "stablehlo_xla" types work: https://github.com/openxla/iree/blob/861815cfe485592dd642088147cd01f16bb39951/compiler/plugins/input/StableHLO/stablehlo-iree/PluginRegistration.cpp#L121-L154